src/optimizer/simple_optimizer.cpp

```diff
@@ -1014,26 +1006,14 @@ SimpleOptimizer::CreateHackingNestedLoopJoinPlan(
   auto projection = CreateHackProjection();
   auto schema = CreateHackJoinSchema();
 
-  // LEFT.4 == RIGHT.1
-  expression::TupleValueExpression* left_table_attr_4 =
-      new expression::TupleValueExpression(type::Type::INTEGER, 0, 0);
-
-  expression::TupleValueExpression* right_table_attr_1 =
-      new expression::TupleValueExpression(type::Type::INTEGER, 1, 0);
-
-  std::unique_ptr<const expression::AbstractExpression> join_predicate(
-      new expression::ComparisonExpression(EXPRESSION_TYPE_COMPARE_EQUAL,
-                                           left_table_attr_4,
-                                           right_table_attr_1));
-
   std::vector<oid_t> join_column_ids_left = {0};   // I_I_ID in the result
   std::vector<oid_t> join_column_ids_right = {1};  // S_I_ID in the table
```

Comment:
Yes I understand that. I'm saying even though we store the logical id here, we can still get the physical column id inside [this function](https://github.com/cmu-db/peloton/pull/445/files#diff-eefe560c257a7f54a22b8f0de670bd01R703) with [this field](https://github.com/cmu-db/peloton/blob/master/src/include/planner/index_scan_plan.h#L172). Basically the logical id in NestedLoopPlan stores the offset in IndexScanPlan::column_ids_, and the IndexScanPlan::column_ids_ stores the physical id. That shouldn't require lots of changes.

I think in general it's not good to store an offset with respect to the input of the child executor. That requires extra exposure of information unless we hard-code it. All other executors only refer the logical offset with respect to the output schema of their children, and don't need to know how the input of their children look like. I basically don't want to add that exposure just for this executor.